### DISCUSSION TITLE
Open VSCode and split integrated terminal ?

### DISCUSSION THREAD

**Discussion by Alarid** [johnlindquist/kit#162] [Q&A]
I'm trying to automate the way that I start my workday, and so far I've managed to make a script that launches every app that I need (expect vscode) and a few browser tabs. I'm really amazed by Kit, and I'm trying to push my automation a little further, but I'm struggling a bit for this part.

Basically, the project I'm working on is a multi-app project. I have a workspace configured in VSCode, in which I have various app folders (1 backend, 2 frontends, + others). Every morning, I open this workspace in vscode, open the integrated terminal on the side, in which I launch 3 commands, splitting the terminal between each. Here's what I'm trying to automate, in pseudo-code:

```
Open vscode integrated terminal to the side
Launch a script: `./launch.sh`
Split horizontally
cd backend && yarn dev
Split horizontally
cd frontend && yarn dev
```

I'm not really sure how to do this with Kit, and to be fair I'm not even sure it's possible, so I'd love to hear your insights on this. I'm guessing this could be done with AppleScript, but I am an absolute beginner with this language.

**Comment by johnlindquist** [comment]
So this gets tricky really quickly. AppleScript and menu items are pretty finicky. But here's something to get you started. They also sometimes appear to fail for no reason and timing and focus can be issues. Maybe someday I'll get around to writing VS code extension for some tighter control of launch setups so we can avoid menu juggling:

```js
//new terminal
await applescript(String.raw`
tell application "System Events"
	tell process "Code" to set frontmost to true
	set _item1 to item 1 of menu bar of application process "Code"
	set _terminalMenu to item 9 of menus of _item1
	tell _terminalMenu
		click _terminalMenu
		delay 0.5
		set _newTerminal to menu item 1 of _terminalMenu
		delay 0.5
		click _newTerminal
	end tell
end tell
	`)

//split terminal
await applescript(String.raw`
tell application "System Events"
	tell process "Code" to set frontmost to true
	set _item1 to item 1 of menu bar of application process "Code"
	set _terminalMenu to item 9 of menus of _item1
	tell _terminalMenu
		click _terminalMenu
		delay 0.5
		set _splitTerminal to menu item 2 of _terminalMenu
		delay 0.5
        click _splitTerminal
	end tell
end tell
	`)

await applescript(`
tell application "System Events"
    keystroke "./launch.sh"
end tell
`)
```

You can also `key code 36` to hit "Enter" after keystrokes, but I usually avoid scripting an "Enter" key press. You never know what sort of popup may have gained focus between the time your script starts at "Enter" is hit.

**Reply by Alarid** [reply]
Thanks, this is indeed getting tricky! Unfortunately, my terminal takes a little while to launch, because of some shell configurations, so the split doesn't work (happens too soon, I think). I'll try and make it work later today, thanks a lot for your help.
